pkg/controller/service/servicecontroller.go

```diff
@@ -125,16 +127,33 @@ func New(
 		nodeLister:       nodeInformer.Lister(),
 		nodeListerSynced: nodeInformer.Informer().HasSynced,
 		workingQueue:     workqueue.NewNamedDelayingQueue("service"),
+		endpoints: make(map[string]int),
 	}
 
 	serviceInformer.Informer().AddEventHandlerWithResyncPeriod(
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: s.enqueueService,
 			UpdateFunc: func(old, cur interface{}) {
+				endpointsChanged := false
 				oldSvc, ok1 := old.(*v1.Service)
 				curSvc, ok2 := cur.(*v1.Service)
+				oldEndpoints := s.endpoints[oldSvc.ObjectMeta.Name]
+				newEndpoints := s.getNumberOfEndpoints(oldSvc)
```

Comment:
We may not want to check endpoints when the service is being created first time. Services that gets updated less frequently should see no additional overhead.